ReSwift/CoreTypes/Store.swift

```diff
@@ -91,61 +83,61 @@ public class Store<State: StateType>: StoreType {
+        (_ subscriber: S, selector: ((State) -> SelectedState)?) {
+            if !_isNewSubscriber(subscriber: subscriber) { return }
+
+            subscriptions.append(Subscription(subscriber: subscriber, selector: selector))
+
+            if let state = self.state {
+                subscriber._newState(state: selector?(state) ?? state)
+            }
+    }
+    #else
     public func subscribe<SelectedState, S: StoreSubscriber
         where S.StoreSubscriberStateType == SelectedState>
         (subscriber: S, selector: ((State) -> SelectedState)?) {
-            #if swift(>=3)
-                if !_isNewSubscriber(subscriber: subscriber) { return }
-            #else
-                if !_isNewSubscriber(subscriber) { return }
-            #endif
+            if !_isNewSubscriber(subscriber) { return }
 
             subscriptions.append(Subscription(subscriber: subscriber, selector: selector))
 
             if let state = self.state {
-                #if swift(>=3)
-                    subscriber._newState(state: selector?(state) ?? state)
-                #else
-                    subscriber._newState(selector?(state) ?? state)
-                #endif
+                subscriber._newState(selector?(state) ?? state)
             }
     }
+    #endif
 
+    #if swift(>=3)
+    public func unsubscribe(_ subscriber: AnyStoreSubscriber) {
+        if let index = subscriptions.index(where: { return $0.subscriber === subscriber }) {
+            subscriptions.remove(at: index)
+        }
+    }
+    #else
     public func unsubscribe(subscriber: AnyStoreSubscriber) {
-        #if swift(>=3)
-            if let index = subscriptions.index(where: { return $0.subscriber === subscriber }) {
-                subscriptions.remove(at: index)
-            }
-        #else
-            if let index = subscriptions.indexOf({ return $0.subscriber === subscriber }) {
-                subscriptions.removeAtIndex(index)
-            }
-        #endif
+        if let index = subscriptions.indexOf({ return $0.subscriber === subscriber }) {
+            subscriptions.removeAtIndex(index)
+        }
     }
+    #endif
 
     public func _defaultDispatch(action: Action) -> Any {
-        if isDispatching {
-            // Use Obj-C exception since throwing of exceptions can be verified through tests
-            #if swift(>=3)
-                NSException.raise(
-                    "ReSwift:IllegalDispatchFromReducer" as NSExceptionName,
-                    format: "Reducers may not dispatch actions.",
-                    arguments: getVaList(["nil"]))
-            #else
-                NSException.raise(
-                    "ReSwift:IllegalDispatchFromReducer",
-                    format: "Reducers may not dispatch actions.",
-                    arguments: getVaList(["nil"]))
-            #endif
+        guard !isDispatching else {
+            fatalError("ReSwift:IllegalDispatchFromReducer - Reducers may not dispatch actions.")
         }
```

Comment:
I would prefer not to have the additional complexity and overhead to maintain too. I forgot to comeback and check if it could be made private to prevent it leaking 😬.

I'll remove the change. I originally made the change because I was having a lot of trouble getting XCTest to catch the NSException in Swift 3. I think I might have worked it out now though.